clang/lib/CIR/CodeGen/CIRGenBuiltinAArch64.cpp

```diff
@@ -1607,6 +1599,50 @@ static mlir::Value buildArmLdrexNon128Intrinsic(unsigned int builtinID,
   }
 }
 
+mlir::Value buildAArch64NeonCall(unsigned int builtinID, CIRGenFunction &cgf,
+                                 llvm::SmallVector<mlir::Type> argTypes,
+                                 llvm::SmallVector<mlir::Value, 4> args,
+                                 llvm::StringRef intrinsicName,
+                                 mlir::Type funcResTy, mlir::Location loc,
+                                 bool isConstrainedFPIntrinsic = false,
+                                 unsigned shift = 0, bool rightshift = false) {
+  // TODO: Consider removing the following unreachable when we have
+  // metaDataTypes and buildConstrainedFPCall features implemented
+  assert(!MissingFeatures::metaDataTypes());
+  assert(!MissingFeatures::buildConstrainedFPCall());
+  if (isConstrainedFPIntrinsic)
+    llvm_unreachable("isConstrainedFPIntrinsic NYI");
+  // TODO: Remove the following unreachable and call it in the loop once
+  // there is an implementation of buildNeonShiftVector
+  assert(!MissingFeatures::metaDataTypes());
+  if (shift > 0)
+    llvm_unreachable("Argument shift NYI");
+
+  if (builtinID != clang::NEON::BI__builtin_neon_vrndns_f32)
+    llvm_unreachable("NYT");
+
+  CIRGenBuilderTy &builder = cgf.getBuilder();
+  for (unsigned j = 0; j < argTypes.size(); ++j) {
+    if (isConstrainedFPIntrinsic) {
+      assert(!MissingFeatures::metaDataTypes());
+    }
+    if (shift > 0 && shift == j) {
+      assert(!MissingFeatures::buildNeonShiftVector());
```

Comment:
I guess our internal testcases are not hitting this path yet, right?

Reply:
right, not yet, having their just to maintain similar skeleton to OG.